### PR TITLE
Fixed config.py

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#19193

*Description of changes:*
- Fixed "config.py" and "vaults.gcp.env" for GCP

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/config.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ class _Config:  # pylint: disable=too-few-public-methods
     JWT_OIDC_ALGORITHMS = os.getenv("JWT_OIDC_ALGORITHMS")
     JWT_OIDC_JWKS_URI = os.getenv("JWT_OIDC_JWKS_URI")
     JWT_OIDC_ISSUER = os.getenv("JWT_OIDC_ISSUER")
-    JWT_OIDC_AUDIENCE = os.getenv("KEYCLOAK_CLIENT_ID")
+    JWT_OIDC_AUDIENCE = os.getenv("JWT_OIDC_AUDIENCE")
```
Comment (JazzarKarim) [inline legal-api/src/legal_api/config.py:87]: @argush3 According to Patrick, hopefully this should do the trick and fix the authentication issue.

File: legal-api/devops/vaults.gcp.env [hunk 2]
```diff
@@ -11,6 +11,7 @@ DATABASE_PASSWORD="op://database/$APP_ENV/business-db/DATABASE_PASSWORD"
 DATABASE_PORT="op://database/$APP_ENV/business-db/DATABASE_PORT"
 DATABASE_UNIX_SOCKET="op://database/$APP_ENV/business-db/DATABASE_UNIX_SOCKET"
 DATABASE_USERNAME="op://database/$APP_ENV/business-db/DATABASE_USERNAME"
+JWT_OIDC_AUDIENCE ="op://keycloak/$APP_ENV/entity-services/ENTITY_SERVICES_CLIENT_ID"
```
Comment (argush3) [inline legal-api/devops/vaults.gcp.env:14]: is this right?  is Patrick going to add this?  I don't see it in 1pass

Comment (JazzarKarim) [reply legal-api/devops/vaults.gcp.env:14]: You don't see it as of now right? Going to ask him.

Comment (argush3) [reply legal-api/devops/vaults.gcp.env:14]: yeah, i see no section called "entity-services" under the `keycloak` vault

Comment (JazzarKarim) [reply legal-api/devops/vaults.gcp.env:14]: MicrosoftTeams-image (5)

It should be in the keyclock vault according to him

Comment (argush3) [reply legal-api/devops/vaults.gcp.env:14]: ok, figure out where i got confused.  there is a "dev-gcp" category under the `keycloak` vault.  it's not actually being used but i was looking for the config there.  

we should be good here.

**Review by argush3 — APPROVED** [review]

File: legal-api/devops/vaults.gcp.env (hunk 2 above)
Comment (JazzarKarim) [reply legal-api/devops/vaults.gcp.env:14]: Deal, sounds good.

**Pull request merged**